### PR TITLE
Case 19724: If sorting PAL by loudness re-sort upon loudness updates

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
Ticket: https://highfidelity.manuscript.com/f/cases/19724/

The ticket covers the list not being sorted on first bringing up the PAL but I noticed the table was not being sorted when a loudness update was received from the js side, unlike other updates. This simple change adds that.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17080/HighFidelity-Beta-PR14864-a2da592.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17080/HighFidelity-Beta-PR14864-a2da592.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17080/HighFidelity-Beta-Interface-PR14864-a2da592.dmg

**Comment by SimonWalton-HiFi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17089/HighFidelity-Beta-PR14864-a2da592.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17089/HighFidelity-Beta-PR14864-a2da592.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17089/HighFidelity-Beta-PR14864-a2da592.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17089/HighFidelity-Beta-Interface-PR14864-a2da592.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17099/HighFidelity-Beta-PR14864-58d8945.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17099/HighFidelity-Beta-PR14864-58d8945.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17099/HighFidelity-Beta-Interface-PR14864-58d8945.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17099/HighFidelity-Beta-PR14864-58d8945.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17099/HighFidelity-Beta-Interface-PR14864-58d8945.exe

File: interface/resources/qml/hifi/Pal.qml [hunk 1]
```diff
@@ -1247,6 +1249,11 @@ Rectangle {
                     }
                 }
             }
+            if (nearbyTable.sortIndicatorColumn == 0 && Date.now() - pal.loudSortTime >= pal.kLOUD_SORT_PERIOD_MS) {
```
Comment (howard-stearns) [inline interface/resources/qml/hifi/Pal.qml]: Not a big deal, but going forward, we should generally use === in javascript. Or, if there's really a reason for using ==, include a comment explaining why.  (== does type casting in often surprising ways)

Comment (SimonWalton-HiFi) [reply interface/resources/qml/hifi/Pal.qml]: Sorry, I keep forgetting this stuff in scripting.

**Comment by SimonWalton-HiFi** [comment]
No, the sort will update at a rate of 2 Hz while sort-by-loud is selected (as long as the PAL is getting updates fast enough). The rate is set by one of the new properties.

**Comment by howard-stearns** [comment]
Doh! ok, I think that's an idea worth trying. 

I am concerned that people already don't understand why they have to press the reset button, and if we have auto updates for one use case but not others, people will get _really_ confused. Is there something we can do to mitigate that confusion? For example, suppose that every time it updated, it highlighted the update button and the loudness column heading, as if they were being pressed, so that it shows people what it is doing.

I don't know. Maybe talk over some ideas with Joshua? If he's ok with it as is, or with what's here plus some mitigation, then I'm ok with it.

**Comment by SimonWalton-HiFi** [comment]
OK - I thought from the code that re-sorting occurred on other updates, but probably the other changes are rare.

**Comment by howard-stearns** [comment]
DNM until we decide how to proceed

**Comment by BridieHiFi** [comment]
I spoke with Joshua and he's in favor of proceeding. See comment in Manuscript:
'Let's move forward with the testing of a 'live refresh' at 10Hz and see if that feels too fast to be useful.'

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17190/HighFidelity-Beta-PR14864-4f2223c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17190/HighFidelity-Beta-PR14864-4f2223c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17190/HighFidelity-Beta-Interface-PR14864-4f2223c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17190/HighFidelity-Beta-PR14864-4f2223c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17190/HighFidelity-Beta-Interface-PR14864-4f2223c.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17207/HighFidelity-Beta-PR14864-73296aa.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17207/HighFidelity-Beta-PR14864-73296aa.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17207/HighFidelity-Beta-Interface-PR14864-73296aa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17207/HighFidelity-Beta-PR14864-73296aa.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17207/HighFidelity-Beta-Interface-PR14864-73296aa.exe

**Comment by BridieHiFi** [comment]
Josh approved, let's get this in please/thank you!

**Pull request merged**